Sources/PointFree/StripeHook.swift

```diff
@@ -1,8 +1,67 @@
-//
-//  StripeHook.swift
-//  PointFree
-//
-//  Created by Stephen Celis on 1/4/18.
-//
-
+import Either
 import Foundation
+import HttpPipeline
+import Prelude
+
+let stripeSubscriptionWebhookMiddleware: Middleware<StatusLineOpen, ResponseEnded, Stripe.Event<Stripe.Subscription>, Data> =
+  validateStripeSignature
+    <| updateStripeSubscription
+
+private let tolerance: TimeInterval = 5 * 60
+
+private func validateStripeSignature<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, A, Data>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, A, Data> {
+
+    return { conn in
+      let pairs = conn.request.value(forHTTPHeaderField: "Stripe-Signature").map {
+        $0.split(separator: ",")
+          .flatMap { pair -> (String, String)? in
+            let pair = pair.split(separator: "=", maxSplits: 1).map(String.init)
+            return tuple <¢> pair.first <*> (pair.count == 2 ? pair.last : nil)
+        }
+      }
+      let components = pairs
+        .map(Dictionary.init)
+        .flatMap { tuple3 <¢> $0["v1"] <*> $0["t"].flatMap(Int.init) <*> conn.request.httpBody }
```

Comment:
should `conn.request.httpBody` be a string instead of data?

Reply:
Realized this (and more) during your review. Addressed!